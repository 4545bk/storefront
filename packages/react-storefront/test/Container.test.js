/**
 * @license
 * Copyright Â© 2017-2018 Moov Corporation.  All rights reserved.
 */
import React from 'react'
import { mount } from 'enzyme'
import Container from '../src/Container'

describe('Container', () => {

  it('renders with no props', () => {
    const component = (
      <Container/>
    )

    expect(mount(component)).toMatchSnapshot()
  })

  it('accepts a className prop', () => {
    const component = (
      <Container className="myClassName"/>
    )

    expect(mount(component)).toMatchSnapshot()
  })

  it('accepts a root class', () => {
    const component = (
      <Container classes={{ root: 'myRootClass' }}/>
    )

    expect(mount(component)).toMatchSnapshot()
  })

  it('spreads props to the underlying div', () => {
    const component = (
      <Container style={{ backgroundColor: 'red' }}/>
    )

    expect(mount(component)).toMatchSnapshot()
  })
  
})