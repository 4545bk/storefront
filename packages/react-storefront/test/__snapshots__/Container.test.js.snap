// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container accepts a className prop 1`] = `
<WithStyles(Container) className="myClassName">
  <Container className="myClassName" classes={{...}} inset={true}>
    <div className="RSFContainer-root-1 myClassName RSFContainer-inset-2" />
  </Container>
</WithStyles(Container)>
`;

exports[`Container accepts a root class 1`] = `
<WithStyles(Container) classes={{...}}>
  <Container classes={{...}} inset={true}>
    <div className="RSFContainer-root-1 myRootClass RSFContainer-inset-2" />
  </Container>
</WithStyles(Container)>
`;

exports[`Container renders with no props 1`] = `
<WithStyles(Container)>
  <Container classes={{...}} inset={true}>
    <div className="RSFContainer-root-1 RSFContainer-inset-2" />
  </Container>
</WithStyles(Container)>
`;

exports[`Container spreads props to the underlying div 1`] = `
<WithStyles(Container) style={{...}}>
  <Container style={{...}} classes={{...}} inset={true}>
    <div className="RSFContainer-root-1 RSFContainer-inset-2" style={{...}} />
  </Container>
</WithStyles(Container)>
`;
